Better warning message for Vis and VisList

cjachekang opened the pull request:
Issue: https://github.com/lux-org/lux/issues/116

![image](https://user-images.githubusercontent.com/47467363/98302386-222be680-1f71-11eb-8ba8-8b6b569cf9f6.png)

dorisjlee commented:
Thanks @cjachekang , I made some minor tweaks to the warning message and added a test. However, some of the test cases is not working, and are resulting in other errors, instead of throwing the custom error.

Example 1: 

```
Vis(["SATAverage","?"],df)
```

Example 2:
```
Vis(["SATAverage","AverageCost","Region=New England|Southeast"],df)
```
Could you see if we could get these to show the custom error? We can add  some more examples into `test_multi_vis`.

cjachekang commented:
Added those two cases to the tests. I moved the SyntaxError check to initialization - probably better to catch the error before all the computations are done.

Also removed traceback for this error so that it looks cleaner - let me know if you want me to add it back.
